### PR TITLE
[torch] Additional folders for shape computations

### PR DESCRIPTION
**Pull request by rsuderman**
A handful of operations are commonly used in shape calculations (slice, concat, broadcast). Added these additional folders to better propagate simple shape computations.

### CODE REVIEW

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -2900,11 +2900,55 @@ OpFoldResult AtenSubIntOp::fold(FoldAdaptor adaptor) {
...
+    return list.getElements()[0];
+
+  auto resultTy = dyn_cast<ValueTensorType>(getType());
+  if (!resultTy || !resultTy.hasSizes() || !resultTy.hasDtype())
     return nullptr;
-  return list.getElements()[0];
+
+  auto bResultTy = resultTy.toBuiltinTensor();
+  if (!bResultTy.hasStaticShape() || bResultTy.getNumElements() > 16)
+    return nullptr;
```
Comment (ScottTodd) [inline lib/Dialect/Torch/IR/TorchOps.cpp:2916]: Why 16 elements? Is that arbitrary, or is there a specific reason?

File: test/Dialect/Torch/canonicalize.mlir [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: torch-mlir-opt %s -canonicalize | FileCheck %s
+// RUN: torch-mlir-opt %s -canonicalize --split-input-file | FileCheck %s
```
Comment (ScottTodd) [inline test/Dialect/Torch/canonicalize.mlir:1]: Oof.. this file is 2000+ lines and wasn't using `--split-input-file`? Can you add the `// -----` blocks to the rest of the file? Maybe in a separate PR to keep history cleaner.

File: lib/Dialect/Torch/IR/TorchOps.cpp (hunk 1 above)
Comment (rsuderman) [reply lib/Dialect/Torch/IR/TorchOps.cpp]: Somewhat arbitrary - its a reasonable upper bound for shape computations.

File: test/Dialect/Torch/canonicalize.mlir (hunk 2 above)
Comment (rsuderman) [reply test/Dialect/Torch/canonicalize.mlir:1]: Follow up would make more sense. We have been incrementally inserting them as models grow. Alternatively someone could write a script to quickly add.

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**